### PR TITLE
Feature: minimum version check

### LINKED ISSUE #1573: Add a node.js min version check
People see for example `Error: Cannot find module 'fs/promises'` essentially random error messages when they upgrade and have an ancient Node.js.

We should add a minimum Node.js version check in https://github.com/SignalK/signalk-server/blob/master/bin/signalk-server with a clear prompt to upgrade as an error message.

### PR DESCRIPTION
**Pull request by JacoKoster**
Fixes #1573

### CODE REVIEW

File: bin/signalk-server [hunk 1]
```diff
@@ -16,6 +16,16 @@
  * limitations under the License.
 */
 
+const { engines } = require('../package.json');
+const semver = require('semver');
+
+const version = engines.node;
+
+if (!semver.satisfies(process.version, version)) {
```
Comment (tkurki) [inline bin/signalk-server]: This is a nice idea in theory, but in practise the minimum is now 16 (we are running tests with it so we don't accidentally mess that up).

I think this should
- !== 18 => log "The recommended version of Node.js is 18, you are running ${process.version}"
- <16  => error log what this does now

File: bin/signalk-server [hunk 2]
```diff
@@ -16,6 +16,16 @@
...
+  console.log(`The installed version of node (${process.version}) is older than the required minimum version (${version}). See https://github.com/SignalK/signalk-server/wiki/Updating-to-Node.js-18 for more information how to upgrade.`);
```
Comment (tkurki) [inline bin/signalk-server]: I reorganised the wiki, this should point to non version specific instructions at https://github.com/SignalK/signalk-server/wiki/Installing-and-Updating-Node.js

**Review by tkurki — CHANGES_REQUESTED** [review]

File: bin/signalk-server (hunk 1 above)
Comment (JacoKoster) [reply bin/signalk-server]: Looking at when 16 goes out of maintenance, it might be worthwhile to make a new issue to test and upgrade to 18?

Comment (tkurki) [reply bin/signalk-server]: Yep, was thinking about the same thing. Docker images already use 18.

But going forward we’ll probably still have minimum separate from recommended that would be latest lts.

Comment (tkurki) [reply bin/signalk-server]: #1584

**Review by tkurki — APPROVED** [review]

**Comment by tkurki** [comment]
Fyi @sailoog @mgrouch

**Comment by mgrouch** [comment]
@tkurki 

BBN Marine OS is already with node 18 and SignalK 2.0

**Comment by sailoog** [comment]
Thanks for notifying. We already changed from node 16 to node 18 in openplotter-signalk-installer v3.2.0. Any existing node 16 installation is converted to a node 18 installation silently. Any queryable setting would help automate this process, any ideas?

**Comment by tkurki** [comment]
New downloads being on node 18 is one thing. There’s also existing installations: if somebody just updates their existing setup’s sk their system will break if min nodejs check is not met. Not that this PR changes much: older versions will break anyway, this just gives a nicer log message.

**Comment by sailoog** [comment]
> What would like to query?

recommended node version

> New downloads being on node 18 is one thing. There’s also existing installations: if somebody just updates their existing setup’s sk their system will break if min nodejs check is not met. Not that this PR changes much: older versions will break anyway, this just gives a nicer log message.

yes, and that is solved in openplotter v3 because we decide when to publish a new openplotter-signalk-installer package containing the new recommended node version. Node and sk will update without breaking anything but it could take some days until we realize that you changed the recommended version, so a system to check this could be nice to automatically update existing systems.

**Comment by tkurki** [comment]
I think the recommendation changes so infrequently that we can do with just keeping syncing between people. And recommendation not being 100% up to date is not crucial.

**Pull request merged**